### PR TITLE
FEAT: Persistent volume along with CD/CD pipeline

### PR DESCRIPTION
**Pull request by shehram-tahir**
### This PR depends on Secrets:
1. VPS_HOST 
2. VPS_SSH_KEY 
3. VPS_USERNAME

### CODE REVIEW

File: .env
Comment (abdullahalhoothy) [inline .env:1]: rename this file to 
storage/secrets/secrets_database.env

File: docker-compose.yml [hunk 1]
```diff
@@ -3,24 +3,22 @@ version: '3.8'
...
+    image: postgres:17
     restart: always
+    container_name: postgres
     environment:
-      POSTGRES_USER: user
-      POSTGRES_PASSWORD: password
-      POSTGRES_DB: mydatabase
+      POSTGRES_USER: ${POSTGRES_USER}
+      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
+      POSTGRES_DB: ${POSTGRES_DB}
```
Comment (abdullahalhoothy) [inline docker-compose.yml:20]: use something like this instead
    env_file:
      - storage/secrets/secrets_database.env

File: docker-compose.yml [hunk 2]
```diff
@@ -3,24 +3,22 @@ version: '3.8'
...
     ports:
       - "5432:5432"
     volumes:
-      - postgres_data:/var/lib/postgresql/data
-
-volumes:
-  postgres_data:
+      - ./my_postgres_data:/var/lib/postgresql/data
```
Comment (abdullahalhoothy) [inline docker-compose.yml:24]: add pgadmin

File: .github/workflows/deployment.yml
Comment (abdullahalhoothy) [inline .github/workflows/deployment.yml:1]: this is fine for now , but in the future this needs to look more like my cicd with all the other secrets

**Review by abdullahalhoothy — APPROVED** [review]

File: .env
Comment (shehram-tahir) [reply .env:1]: No, this .env is solely for by docker-compose by default to set variables in the environment. It is better to keep the configs together
We can make sections in this file for different configurations if you want segregation.

For example: We could classify the variables using comment sections like the below:

// Database variables: 
POSTGRES_USER: postgres
POSTGRES_PASSWORD: postgres
POSTGRES_DB: mydatabase

// Backend variables:
variable_one:

// Frontend Variables
variable_one:

File: docker-compose.yml (hunk 2 above)
Comment (shehram-tahir) [reply docker-compose.yml:24]: okay, I will add in the next PR

**Pull request merged**